athena-gcs/src/test/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandlerTest.java

```diff
@@ -0,0 +1,319 @@
+        table.setCatalogId(CATALOG);
+        List<Column> columns = List.of(
+                createColumn("name", "String")
+        );
+        table.setPartitionKeys(columns);
+        GetTableResult getTableResult = new GetTableResult();
+        getTableResult.setTable(table);
+        PowerMockito.when(awsGlue.getTable(any())).thenReturn(getTableResult);
+        StorageMetadata storageMetadata = mock(StorageMetadata.class);
+        Whitebox.setInternalState(gcsMetadataHandler, storageMetadata, storageMetadata);
+        PowerMockito.when(storageMetadata.buildTableSchema(any(), any())).thenReturn(schema);
+        GetTableResponse res = gcsMetadataHandler.doGetTable(blockAllocator, getTableRequest);
+        Field expectedField = res.getSchema().findField("name");
+        assertEquals(Types.MinorType.VARCHAR, Types.getMinorTypeForArrowType(expectedField.getType()));
+    }
+
+    @Test
+    public void testGetPartitions() throws Exception
+    {
+        environmentVariables.set("glue_catalog", "12345678910");
+        Field field = new Field("year", FieldType.nullable(new ArrowType.Utf8()), null);
+        Schema schema = SchemaBuilder.newBuilder().addField("id", new ArrowType.Int(64, false)).build();
+        Table table = new Table();
+        table.setName("birthday");
+        table.setDatabaseName("mydatalake1");
+        table.setParameters(ImmutableMap.of("classification", "parquet",
+                "partition.pattern", "year={year}/birth_month{month}/{day}")
+        );
+        table.setStorageDescriptor(new StorageDescriptor()
+                .withLocation("gs://mydatalake1test/birthday/").withColumns(new Column()));
+        table.setCatalogId(CATALOG);
+        List<Column> columns = List.of(
+                createColumn("year", "varchar"),
+                createColumn("month", "varchar"),
+                createColumn("day", "varchar")
+        );
+        table.setPartitionKeys(columns);
+        GetTableResult getTableResult = new GetTableResult();
+        getTableResult.setTable(table);
+        PowerMockito.when(awsGlue.getTable(any())).thenReturn(getTableResult);
+        GetTableLayoutRequest getTableLayoutRequest = Mockito.mock(GetTableLayoutRequest.class);
+        Mockito.when(getTableLayoutRequest.getTableName()).thenReturn(new TableName("mydatalake1", "birthday"));
+        Mockito.when(getTableLayoutRequest.getCatalogName()).thenReturn("fakedatabase");
+        Mockito.when(getTableLayoutRequest.getSchema()).thenReturn(schema);
+        Constraints constraints = new Constraints(createSummaryWithLValueRangeEqual("id", new ArrowType.Int(64, false), 1L));
+        Mockito.when(getTableLayoutRequest.getConstraints()).thenReturn(constraints);
+        BlockWriter blockWriter = Mockito.mock(BlockWriter.class);
+        gcsMetadataHandler.getPartitions(blockWriter, getTableLayoutRequest, queryStatusChecker);
```

Comment:
It looks like this test is missing some assertions for the response of this call right?

Reply:
added assertion.